src/config.rs

```diff
@@ -5,24 +5,27 @@ use std::io::{BufReader, BufWriter, Read};
 use std::path::Path;
 
 fn build(ctx: &Contexts, ns: Option<&str>, strbuf: &str) -> KubeConfig {
-    let mut config: KubeConfig = serde_yaml::from_str(&strbuf).unwrap();
+    let mut config: KubeConfig = serde_yaml::from_str(strbuf).unwrap();
     config.api_version = "v1".to_string();
     config.kind = "Config".to_string();
-    config.current_context = format!("{}", ctx.name);
-
-    let ns = if ns.is_some() {
-        ns.unwrap().to_string()
-    } else if config.contexts.len() > 0 && !config.contexts[0].context.namespace.is_empty() {
-        config.contexts[0].context.namespace.to_string()
-    } else if !ctx.context.namespace.is_empty() {
-        ctx.context.namespace.to_string()
-    } else {
-        "default".to_string()
+    config.current_context = ctx.name.to_string();
+
+    let ns = match ns {
+        Some(namespace) => namespace.to_string(),
+        None => {
+            if !config.contexts.is_empty() && !config.contexts[0].context.namespace.is_empty() {
```

Comment:
Thank you! That function wasn't the prettiest one 😅

Reply:
No worries, it was correct after all :smile: